### PR TITLE
fix(reports): improve system usage stats

### LINKED ISSUE #5148: Poor error handling in System Usage Statistics report form
In Hospital > Reports > System Usage Statistics

if you click the [Preview] button before selecting a date, you get this error message in red at the top of the page:
    `Your request could not be translated into valid SQL. Please modify your request and try again.`

A more informative error message would be better.

### PR DESCRIPTION
**Pull request by jniles**
Improved error handling on the system usage statistics report.

Closes #5148.

### CODE REVIEW

File: client/src/modules/reports/generate/system_usage_stat/system_usage_stat.html [hunk 1]
```diff
@@ -27,6 +27,7 @@ <h3 class="text-capitalize" translate>REPORT.SYSTEM_USAGE_STAT.TITLE</h3>
               max-date = "ReportConfigCtrl.today"
               date-value="ReportConfigCtrl.reportDetails.date"
               on-change="ReportConfigCtrl.onDateChange(date)"
+              limit-min-fiscal
```
Comment (jmcameron) [inline client/src/modules/reports/generate/system_usage_stat/system_usage_stat.html:30]: What does this do?

**Review by jmcameron — COMMENTED** [review]
The code looks good.  The fix works properly and addresses the issue.
I had one suggestion.  At the top of the report, I see the message:
"This report produce, for a given period, stats of, Total of patients recorded by hour of the day..."

The rest of the message makes it clear that the report is for a single day, but the term "given period" is confusing.  Perhaps it should say:  "... for the selected day, ...".

I tested it with the Vanga db and was able to see the table for some dates in July.

File: client/src/modules/reports/generate/system_usage_stat/system_usage_stat.html (hunk 1 above)
Comment (jniles) [reply client/src/modules/reports/generate/system_usage_stat/system_usage_stat.html:30]: Limits the minimum date to the first fiscal year date.

**Comment by jniles** [comment]
@jmcameron changes landed.

File: client/src/i18n/en/report.json [hunk 2]
```diff
@@ -114,7 +114,7 @@
     },
     "SYSTEM_USAGE_STAT" : {
       "TITLE" : "System Usage Statistics",
-      "DESCRIPTION" : " This report  produce, for a given period, stats of, Total of patients recorded by hour of the day, Total of invoices made by hour of the day and total of cash payments made by hour of the day"
+      "DESCRIPTION" : " This report produces statistics on the total patients, invoices, and cash payments made by hour of the day for a given date."
```
Comment (jmcameron) [inline client/src/i18n/en/report.json:117]: 👌

**Review by jmcameron — APPROVED** [review]
Looks good.   I'm going ahead and approving it.   

However, as I looked at it, I wondered if you want to include more columns of statistics.  Eg:  total $ paid by patients, value of stock used in the day, number of admissions, discharges, births, deaths, etc.  Perhaps these should be added in another Issue.

**Comment by jniles** [comment]
@jmcameron I don't want to put that in this PR if that is okay.  It was originally just supposed to be a bug fix.

 Can you add your thoughts to https://github.com/IMA-WorldHealth/bhima/issues/5848?

**Comment by jniles** [comment]
This has approval, so in it goes.

bors r+

**Comment by jmcameron** [comment]
> Can you add your thoughts to #5848?

Done

**Pull request merged**